### PR TITLE
feat(dlp): added a sample for inspect hotword rule 

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect hot word rule](https://cloud.google.com/dlp/docs/samples/dlp-inspect-hotword-rule) with a unit test case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// inspectWithHotWordRules inspects data with hot word rule, it uses custom
+// regex with a hot word rule to increase the likelihood match
+func inspectWithHotWordRules(w io.Writer, projectID, textToInspect, customRegexPattern, hotWordRegexPattern string) error {
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_hotword_rule.go:29]: nit: consider reducing the number of arguments passed into the sample. It's okay to hard-code some dummy data for demonstration purposes.

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_hotword_rule.go:49]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Construct the custom regex detectors
+	var customInfoType = &dlppb.CustomInfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_hotword_rule.go:59]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		Type: &dlppb.CustomInfoType_Regex_{
+			Regex: &dlppb.CustomInfoType_Regex{
+				Pattern: customRegexPattern,
+			},
+		},
+		Likelihood: dlppb.Likelihood_POSSIBLE,
+	}
+
+	// Construct hotword rule.
+	var inspectionRuleSet = &dlppb.InspectionRuleSet{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_hotword_rule.go:72]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+				inspectionRuleSet,
+			},
+			IncludeQuote: true,
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_hotword_rule.go:115]: issue: only return the error. The caller can inspect the error to understand what happened.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/inspect/inspect_hotword_rule.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

File: dlp/snippets/inspect/inspect_hotword_rule.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	var inspectionRuleSet = &dlppb.InspectionRuleSet{
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_HotwordRule{
+					HotwordRule: &dlppb.CustomInfoType_DetectionRule_HotwordRule{
+						HotwordRegex: &dlppb.CustomInfoType_Regex{
+							Pattern: hotWordRegexPattern,
+						},
+						Proximity: &dlppb.CustomInfoType_DetectionRule_Proximity{
+							WindowBefore: int32(10),
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

File: dlp/snippets/inspect/inspect_hotword_rule.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

File: dlp/snippets/inspect/inspect_hotword_rule.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

File: dlp/snippets/inspect/inspect_hotword_rule.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

File: dlp/snippets/inspect/inspect_hotword_rule.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_hotword_rule.go]: fixed it.

**Review by telpirion — APPROVED** [review]

**Review by annadata — APPROVED** [review]
LGTM.

**Review by telpirion — APPROVED** [review]

**Pull request merged**